src/frontegg-auth/src/error.rs

```diff
@@ -7,35 +7,56 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use std::sync::Arc;
 use thiserror::Error;
 
 use crate::AppPasswordParseError;
 
-#[derive(Error, Debug)]
+#[derive(Clone, Error, Debug)]
 pub enum Error {
     #[error(transparent)]
     InvalidPasswordFormat(#[from] AppPasswordParseError),
     #[error("invalid token format: {0}")]
     InvalidTokenFormat(#[from] jsonwebtoken::errors::Error),
     #[error("authentication token exchange failed: {0}")]
-    ReqwestError(#[from] reqwest::Error),
+    ReqwestError(Arc<reqwest::Error>),
```

Comment:
These are being Arc'd to allow for sending the same error to all inflight requests and avoid a bunch of clones?

Reply:
I wrapped some error types in Arcs because they surprisingly can't be Clone-d. I was think of wrapping the entire `Error` in an Arc, but that changes the external API which I didn't really want to do